src/Context.php

```diff
@@ -234,6 +234,21 @@ public function withName(string $name): self
         );
     }
 
+    public function getPath(?string $path = null): string
```

Comment:
I don't really like this API. As a user I wouldn't expect to have an argument to a getter method. More over, the argument mutate the return value, but not the property.

More over, the `str_starts_with() call is suspicious. 

Finally, This can be done in userland with a regular string concatenation.

Could you remove the suffix behavior 

Finally, the name is weird to me. `getPath()` returns the currentDirectory, so let's call it `getCurrentDirectory()`.

Reply:
Yes, that's why I asked about the naming - it's confusing ^^

So if the method returns the `currentDirectory`, the PR doesn't really make sense any more and closing it, and so use public property directly?

The aim was to have access to a quick helper to have a path concat in relation to the context's `currentDirectory`.

Reply:
Yes I think it can be closed